tests/rptest/tests/schema_registry_test.py

```diff
@@ -1223,6 +1223,122 @@ def move_leader():
                 check_connection(n.account.hostname)
             move_leader()
 
+    @cluster(num_nodes=3)
+    def test_restart_schema_registry(self):
+        # Create several schemas on one subject and return the list schemas and schema ids
+        def register_schemas(subject: str):
+            schemas = [
+                json.dumps({"schema": schema1_def}),
+                json.dumps({"schema": schema2_def}),
+                json.dumps({"schema": schema3_def})
+            ]
+            schema_ids = []
+
+            for schema in schemas:
+                self.logger.debug(f"Post schema {json.loads(schema)}")
+                result_raw = self._post_subjects_subject_versions(
+                    subject=subject, data=schema)
+                self.logger.debug(result_raw)
+                assert result_raw.status_code == requests.codes.ok
+                res = result_raw.json()
+                self.logger.debug(res)
+                assert "id" in res
+                schema_ids.append(result_raw.json()["id"])
+
+            return schemas, schema_ids
+
+        # Given a list of subjects, check that they exist on the registry
+        def check_subjects(subjects: list[str]):
+            result_raw = self._get_subjects()
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_subjects = result_raw.json()
+            self.logger.debug(res_subjects)
+            assert type(res_subjects) == type([])
+            res_subjects.sort()
+            subjects.sort()
+            assert res_subjects == subjects
+
+        # Given the subject and list of ids, check that the ids match
+        def check_subject_versions(subject: str, schema_ids: list[int]):
+            result_raw = self._get_subjects_subject_versions(subject=subject)
+            self.logger.debug(result_raw)
+            assert result_raw.status_code == requests.codes.ok
+            res_ids = result_raw.json()
+            self.logger.debug(res_ids)
+            assert type(res_ids) == type([])
+            res_ids.sort()
+            schema_ids.sort()
+            assert res_ids == schema_ids
+
+        # Given the subject, list of schemas, and list of ids, check the schema that maps to the id
+        def check_each_schema(subject: str, schemas: list[str],
+                              schema_ids: list[int]):
+            for schema_id in schema_ids:
+                self.logger.debug(
+                    f"Fetch schema {schema_id} on subject {subject}")
+                result_raw = self._get_subjects_subject_versions_version(
+                    subject=subject, version=schema_id)
```

Comment:
It's a bit confusing and possibly fragile to assume that the version and the schema ID are equivalent.